### ISSUE TITLE
Implementation of simple InstanceScope

### ISSUE THREAD

**Issue by davebirch** [alecthomas/injector#147]
Am aware this is more a question, rather than a bug, but perhaps an indication that an example scope implementation (other that an example of seemingly re-implementing NoScope would be helpful for those new to injector.

Anyway, I am trying to implement a simple custom scope which provides the same instance of a given class to any sub-objects in that class that might require a reference to it.

**Comment by jstasiak** [comment]
Yeah, the documentation can be improved in this regard. About your case – can you provide some pseudo-code of what you're trying to achieve?

**Comment by davebirch** [comment]
Sorry, hit enter too early - edited the OP to include an example.

**Comment by jstasiak** [comment]
I don't think there's a way to do this at the moment. Possibly the scope system could be improved to handle this, but it'd require some redesign I believe.
